### PR TITLE
test: Add a testloop test for issuing view-client requests to an archival node

### PR DESCRIPTION
**Pull request by tayfunelmas**
We are planning to make changes to the archival node storage (eg. eliminate some columns from cold db). For example, see [this draft](https://docs.google.com/document/d/1lfiMus_CBGts-rXTf6p3v0asIN9VmIU5VCwt7dN8fOA/edit?pli=1#heading=h.v8uj1ljdaxd).

But first, we want to make sure we have good coverage of archival node functionality. Today archival nodes answer historical queries about the chain data. It does it using the ViewClient, so we add a TestLoop test to exercise the codepaths in ViewClient to respond to the requests.

Note that the queries are issues through JSON-RPC interface (Restful HTTP API), but we do not need to setup the test at the RPC level. All these RPC requests are forwarded to the ViewClient, so it is sufficient for our goal to check the ViewClient functionality.

For now this PR gives a small number of example test cases (for getting block, chunk, and validator info). We will add more test cases after this PR is merged.

Changes in this PR:
- Add functionality to consider some of the clients as an archival node client (could be validator or non-validator).
- Change the testloop builder to use split-store for archival nodes. Also update the ViewClient actor initialization to use view-specific instances of EpochManager, ShardTracker, and RuntimeAdapter.
- - Add an option to the testloop builder to change the number of epochs to keep when GC'ing (this allows to quickly get to post-GC state to check that archival node preserves the data after GC).
- Add a utility named `ViewRequestSender` to send messages to ViewClient. *Note that* ideally we should run this in async way (send message and wait for the response with await), but testloop does not provide this functionality yet, so we call the `handle` function instead of sending message via `view_client_sender`.
- Remove the option to enable/disable GC (it must always be enabled).

### CODE REVIEW

File: integration-tests/src/test_loop/builder.rs [hunk 1]
```diff
@@ -43,11 +44,15 @@ pub struct TestLoopBuilder {
     test_loop: TestLoopV2,
     genesis: Option<Genesis>,
     clients: Vec<AccountId>,
+    /// Accounts whose clients should be configured as an archival node.
+    /// This should be a subset of the accounts in the `clients` list.
+    archival: HashSet<AccountId>,
```
Comment (pugachAG) [inline integration-tests/src/test_loop/builder.rs:49]: nit: `archival` -> `archival_clients` to have consistent naming with the previous field

File: integration-tests/src/test_loop/tests/view_requests_to_archival_node.rs [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+fn test_view_requests_to_archival_node() {
+    init_test_logger();
+    let builder = TestLoopBuilder::new();
+
+    let initial_balance = 10000 * ONE_NEAR;
+    let accounts = (0..NUM_ACCOUNTS)
+        .map(|i| format!("account{}", i).parse().unwrap())
+        .collect::<Vec<AccountId>>();
+    // Validators take all the roles: block+chunk producer and chunk validator.
+    let validators = (0..NUM_VALIDATORS).map(|idx| accounts[idx].as_str()).collect_vec();
```
Comment (pugachAG) [inline integration-tests/src/test_loop/tests/view_requests_to_archival_node.rs:43]: nit: maybe `accounts.iter().take(NUM_VALIDATORS).map(|acc| acc.as_str()).collect_vec()`

File: integration-tests/src/test_loop/utils/view_client.rs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    /// Gets a block using a [`GetBlock`] request from the view client at index `idx`.
+    pub(crate) fn get_block(
+        &self,
+        request: GetBlock,
+        test_loop: &mut TestLoopV2,
+        idx: usize,
+    ) -> Result<BlockView, Error> {
+        let view_client = test_loop.data.get_mut(&self.handles[idx]);
+        match view_client.handle(request) {
```
Comment (pugachAG) [inline integration-tests/src/test_loop/utils/view_client.rs:35]: nit: you can use `.map_err(|e| e.into())`

**Review by pugachAG — APPROVED** [review]

File: integration-tests/src/test_loop/builder.rs (hunk 1 above)
Comment (tayfunelmas) [reply integration-tests/src/test_loop/builder.rs:49]: done.

File: integration-tests/src/test_loop/tests/view_requests_to_archival_node.rs (hunk 2 above)
Comment (tayfunelmas) [reply integration-tests/src/test_loop/tests/view_requests_to_archival_node.rs:43]: nice, thanks, done.

File: integration-tests/src/test_loop/utils/view_client.rs (hunk 3 above)
Comment (tayfunelmas) [reply integration-tests/src/test_loop/utils/view_client.rs:35]: yes, this is what I was looking for, thanks :)

**Pull request merged**